alf/bin/train.py

```diff
@@ -120,6 +120,14 @@ def training_worker(rank: int, world_size: int, conf_file: str, root_dir: str):
         if world_size > 1:
             # Specialization for distributed mode
             dist.init_process_group('nccl', rank=rank, world_size=world_size)
+            # In distributed mode, this function is running in a newly spawned
+            # process, and will have blank FLAGS at this moment.
+            #
+            # Define and then load the cached flags from the parent process so
+            # that all the functionalities that depends on the abseil FLAGS can
+            # work smoothly and transparently.
+            _define_flags()
+            flags_cache.load(root_dir)
```

Comment:
it seems to me a simpler and more robust solution is to pass the command line as argument to training_worker and parse it here.

Reply:
Right, if we don't have to store it in `root_dir`.

Reply:
> it seems to me a simpler and more robust solution is to pass the command line as argument to training_worker and parse it here.

Turns out that this `sys.argv` is copied even during `spawn`, and I can directly parse `sys.argv` in the sub process.

Will abandon this PR and apply the simpler trick in the other PR.